src/Components/Inputs/Input.tsx

```diff
@@ -247,7 +257,8 @@ export const Input = forwardRef<InputRef, InputProps>(
         />
         {clearElement}
         {type === InputType.Checkbox && <div className={inputCheckboxClass} />}
-        {iconElement}
+        {/** If both icon and colorPreview are set in props, icon has higher priority */
+        !!icon ? iconElement : colorElement}
```

Comment:
is `colorElement` always going to exist? this kind of implies that if `icon` is falsy, `colorElement` will always be shown. In reality, if they both don't exist, nothing will get rendered. But the intent of the code isn't as clear - this makes it seem like if `icon` is falsy, `colorElement` should be always shown.

it might be clearer to do something like this above where `iconElement` and `colorElement` are defined:

```ts
let iconElement = null
if (icon) {
  iconElement = <Icon glyph={icon} className="cf-input-icon" />
} else if (colorPreview) {
  iconElement = <ColorPreview color={colorPreview} className="cf-input-icon" />
}
```

then down here, leaving this as is
```ts
{iconElement}
```

that way, `icon` takes precedence, and the color preview is only shown if icon doesn't exist and `colorPreview` exists.

The outcome will be the same, but the intent of the code is a little bit clearer - that icon takes precedence, and the color preview is only shown when icon doesn't exist and `colorPreview` does exist

Reply:
Thanks! This suggestion makes more sense!